### ISSUE TITLE
Unable to catch the body of error

### ISSUE THREAD

**Issue by oscarjlep** [axios/axios#1369]
#### Summary

I already saw multiple posts regarding this issue, but was still unable to find what the problem really is. I have created generic method for throwing out JSON-errors on Java Spring backend:

```
public static void responseErrorJSON(ResourceResponse response, String errorMessage, String errorCode) throws IOException {
		response.setContentType("application/json");
		ErrorJSON result = new ErrorJSON();
		result.errorMessage = errorMessage;
		result.errorCode = errorCode;
		
		PrintWriter writer = null;
		response.setProperty(response.HTTP_STATUS_CODE, "500");
		writer = response.getWriter();

		writer.write(gson.toJson(result));
	}
```

I'm calling this method as such:

```
return axios.get(context.lr.url.getSalesOrderWithRows)
					.then(function(response) {
						if(typeof callback != "undefined")
							callback(response.data);
					}).catch(function (error) {
						console.log(error);
					});
```

.... but for some reason I'm unable to read message body sent by backend as correct JSON, even though it's displaying just right within Chrome debugger:

![image](https://user-images.githubusercontent.com/26794513/36370865-e2c0929e-1568-11e8-8593-f034100fcacc.png)

Headers look like these, and I assume something here isn't correct as:

![image](https://user-images.githubusercontent.com/26794513/36370827-c0fedc6a-1568-11e8-9553-70a89a3005d9.png)

What I'm getting with console.log(error1,error2) within catch is:

![image](https://user-images.githubusercontent.com/26794513/36371018-84420e9a-1569-11e8-817d-d1ab01b5821a.png)

... so, basically I'm getting generic error-message-string which is generated by axios, but it doesn't seem to contain any object properties such as "error.description" or such, nor error-json generated by backend. That "undefined" is the "error2", which I tried in order to check if it would have container the message body, but apparently doesn't.

#### Context

- axios version: *e.g.: v0.17.0*
- Environment: *e.g.: Liferay 6.2 / Java, chrome 64, windows 10*

**Comment by emilyemorehouse** [comment]
You should definitely be able to access that information. I don't see anything jumping out at me that's incorrect in your response or request info. 

I'm not sure if it will help you or if you've already seen this info, but here's an example test that passes:
```
    var data = {
      reason: 'Invalid something or other',
    };

    server = http.createServer(function (req, res) {
      res.statusCode = 500;
      res.setHeader('Content-Type', 'application/json;charset=utf-8');
      res.end(JSON.stringify(data));
    }).listen(4444, function () {
      axios.get('http://localhost:4444/').catch(function (err) {
        test.deepEqual(err.response.data, data);
        test.equal(err.response.status, 500);
        test.done();
      });
    });
```
You can see that you're able to access the status and data via `res.response.status` or `res.response.data`. 

Can you see if you can access either of those attributes? If you just log `error`, you'll get the actual Error object, but you can drill down further. See this example:
<img width="686" alt="screen shot 2018-02-19 at 9 22 13 am" src="https://user-images.githubusercontent.com/4264897/36387575-8fdbb82c-1556-11e8-8d52-d966852c5ad9.png">

**Comment by oscarjlep** [comment]
I decided to try using node server to send the error in order to count out the possibility of Tomcat doing something weird, so I created following server script:

 ```
   server = http.createServer(function (req, res) {
      res.statusCode = 500;
      res.setHeader('Content-Type', 'application/json;charset=utf-8');
      res.setHeader('Access-Control-Allow-Origin', '*');
      res.setHeader('Access-Control-Allow-Credentials', 'false');
      res.end(JSON.stringify(data));
    }).listen(4444, function () {
    });
```
Then I call this with:
```
return axios.get("http://localhost:4444/", {withCredentials: false})
	.then(function(response) {
		if(typeof callback != "undefined")
			callback(response.data);
	}).catch(function (error) {
		console.log(error);
});
```

... and the outcome is exactly same, un-drillable string! Then I decided to check how it handles with Firefox and it seems that Firefox shows:

![image](https://user-images.githubusercontent.com/26794513/36410149-fc652cb4-1617-11e8-9bef-145dd72cd31a.png)

.... which is drillable!

This felt really strange, so I double checked the existance of error.response and yes, it really DID exist all the time. I must have had done something wrong initially when trying out checking the existance of response. This is very misleading as Chrome for some reason doesn't seem to output as drillable (how is it even able to output object as plain string?).  

I'd suggest considering separating the response object from mixed result object, so that callback function would have optional response object. Developer could write either function(result) or function(result, response).

Thank you for all the help!
